src/gluonts/model/trivial/mean.py

```diff
@@ -60,10 +61,20 @@ def __init__(
 
     def predict(self, dataset: Dataset, **kwargs) -> Iterator[SampleForecast]:
         for item in dataset:
-            mean = np.mean(item["target"][-self.context_length :])
+            if self.context_length is not None:
+                target = item["target"][-self.context_length :]
+            else:
+                target = item["target"]
+
+            mean = np.nanmean(target)
+            std = np.nanstd(target)
```

Comment:
Great, thank you :-) I was about to ask for this in the other PR adding this predictor

Reply:
@lovvge deserves some credit as well for this!